components/hook-request/node/index.mjs

```diff
@@ -31,21 +29,18 @@ export default (dependencies) => {
         ...["request", "get"].map((key) => ({ object: Https, key })),
       ].map(({ object, key }) => ({ object, key, value: object[key] }));
       const spyRequest = (request) => {
-        const index1 = incrementEventCounter(frontend);
-        const index2 = incrementEventCounter(frontend);
+        let index1;
+        let index2;
         request.on("finish", () => {
           const { method, path: url } = request;
           const headers = request.getHeaders();
-          sendEmitter(emitter, recordBeginBundle(frontend, index1, null));
-          sendEmitter(
-            emitter,
-            recordBeforeRequest(frontend, index2, {
-              protocol: "HTTP/1.1",
-              method,
-              url,
-              headers,
-            }),
-          );
+          index1 = recordBeginBundle(agent, null);
+          index2 = recordBeforeRequest(agent, {
+            protocol: "HTTP/1.1",
```

Comment:
Can we detect the actual value here?

Reply:
Are you referring to the protocol? If so, we are hooking to `http` native node module which only handles `HTTP/1.1`.

Reply:
So we could easily get the actual value, but it will always be `"HTTP/1.1". If you still think that is clearer to use the value provided by the request object, we can do that easy.

Reply:
Yes it would be nice to populate from the request object.

Reply:
Okay, I remember doing some digging and it wasn't clear whether `http.Server` module could handle `HTTP/1.0` requests. Anyway, we don't have the problem if we use the actual value.

Reply:
Actually, (outgoing) client requests do not provide a way to inspect the protocol. For the reason I cited before: it is always going to be `HTTP/1.1`. In the doc [ClientRequest](https://nodejs.org/api/http.html#class-httpclientrequest) have a [`protocol`](https://nodejs.org/api/http.html#requestprotocol) field but it is the URL protocol -- ie: `"http://"`. Note that (incoming) server requests do have a [`httpVersion`](https://nodejs.org/api/http.html#messagehttpversion) property because it can either be `1.0` or `1.1`. So in short, the current code does not need to be fixed.